DietPi-Login | Fix and tuning

MichaIng opened the pull request:
**Status**: Testing

**Commit list/description**:
+ DietPi-Login | Fix changed AUTO_START_INDEX => auto_start_index
+ DietPi-Login | Check for PostBoot finish only once before loop: Tiny speed-up of following loops
+ DietPi-Login | Check for AUTO_SETUP_AUTOMATED only once before loop, to have "export G_USER_INPUTS=0" (on invalid install state) not being overwritten on next loop
+ Make above checks work without globals loaded

MichaIng committed:
v6.22

+ DietPi-Login | Fix changed AUTO_START_INDEX => auto_start_index
+ DietPi-Login | Check for PostBoot finish only once before loop: Tiny speed-up of following loops
+ DietPi-Login | Check for AUTO_SETUP_AUTOMATED only once before loop, to have "export G_USER_INPUTS=0" not being overwritten on error

MichaIng commented:
@Fourdee 
> `(( ! $G_USER_INPUTS )) && additional_text='Automated setup is in progress, the system will reboot automatically when completed.'`

So this is not 100% true, if no additional software was selected, right? Simply leave the hint to finish setup on active terminal session? We could even print the TTY/pts where the active update/install is attached to.

Fourdee commented:
@MichaIng 

> So this is not 100% true, if no additional software was selected, right?

Good point 👍 Will take a look.

Commit:
v6.22

#Sudo required for non-root

dietpi@DietPi:~$ systemctl status dietpi-postboot
Failed to connect to bus: No such file or directory

Commit:
v6.22

+ Tweaks

Commit:
v6.22

+ Might also need a failsafe int check

Commit:
v6.22

+ Count \o/

Commit:
v6.22

+ Switch to dietpi-login, to ensure pgrep better match

Fourdee commented:
made some additional changes/tweaks , will merge and test.

MichaIng commented on dietpi/login:
```diff
@@ -189,7 +189,7 @@
 
 				# - Prompt and wait if this script runs in other session already
-				if (( $(pgrep 'login' 2> /dev/null | wc -l) >= 2 )); then
+				if (( $(pgrep -c 'login') >= 2 )); then
```
@Fourdee 
Quick check required if `-c` is available on Jessie as well, e.g. `-i` is not.
Okay Jessie DOES support it 👍: https://manpages.debian.org/jessie/procps/pgrep.1.en.html

MichaIng commented on dietpi/dietpi-login:
```diff
@@ -9,7 +9,7 @@
 	#
 	# Info:
-	# - filename /DietPi/dietpi/login
+	# - filename /DietPi/dietpi/dietpi-login
```
Jep was just thinking the same 👍.
